test/support/factories/alerts/alert.ex

```diff
@@ -28,4 +31,40 @@ defmodule Test.Support.Factories.Alerts.Alert do
       url: Faker.Internet.url()
     }
   end
+
+  def for_route(alert, route_id) do
+    %{
+      alert
+      | informed_entity:
+          InformedEntitySet.build(:informed_entity_set,
+            route: route_id,
+            entities: [
+              InformedEntity.build(:informed_entity, route: route_id)
+            ]
+          )
+    }
+  end
+
```

Comment:
suggestion: If you adjust this function to be its own factory, you wouldn't need to create an alert and pass it to `for_route/2`. Say you want to call `build(:alert_with_route, %{route_id: "Blue"})`, that could work with something like this....

```elixir
def alert_with_route_factory(%{route_id: route_id} = attrs) do
   informed_entity = InformedEntitySet.build(:informed_entity_set,
      route: route_id,
      entities: InformedEntity.build_list(1, :informed_entity, route: route_id)
    )
   build(:alert, informed_entity: informed_entity)
end
```